cpp/ps0008/ps0008_sol.cpp

```diff
@@ -1,34 +1,71 @@
 #include <iostream>
 #include <string>
+#include <cctype>
 
 using namespace std;
+
+// decryption process 2/3
 string decrypt(string ciphertext, int key) {
     string plaintext = "";
     int n = ciphertext.length();
+    key %= 26; // remove multiples of key value
     for (int i = 0; i < n; ++i) {
         char ch = ciphertext[i];
         if (!isalpha(ch)) {
             plaintext += ch;
             continue;
         }
-         ch = tolower(ch);
-        int shifted = ch -'a' + key;
-         if (shifted < 0) {
+        int shifted = tolower(ch) -'a' - key; // you decrept not encrypt
+        if (shifted < 0) {
             shifted += 26;
         }
-     plaintext += (char)(shifted + 'a');
+        else if (shifted > 25) {
+            shifted -= 26;
+        } // if the key is negative the shift is out-range too
+        plaintext += (char)(shifted + (islower(ch) ? 'a' : 'A')); // make the decrepted letter be the same letter case
     }
     return plaintext;
 }

```

Comment:
- [x] defined function takes two parameters (ciphertext and key)for decryption process
- [ ] considered the case of the letter in the message while decryption
- [x] skip numbers and other special characters while decryption
- [ ] considered that the key may be positive or negative value
- [ ] returned the actual message (message before encryption)